package-type-check/pkg/utils/package_utils.go

```diff
@@ -0,0 +1,172 @@
+package utils
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+// IsPackageInstalled checks if the package is installed
+func IsPackageInstalled(pkg string) error {
+	cmd := exec.Command("apk", "info", "-eq", pkg)
+	cmd.Stdout = nil
+	cmd.Stderr = nil
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("package %q is not installed: %w", pkg, err)
+	}
+	return nil
+}
+
+// GetTotalApkCount retrieves the total count of installed APK packages in the environment
+func GetTotalApkCount() int {
+	cmd := exec.Command("apk", "info", "-L")
+	output, err := cmd.Output()
+	if err != nil {
+		return 0
+	}
+
+	// Split the output by lines and count the number of lines
+	lines := strings.Split(string(output), "\n")
+	count := 0
+	for _, line := range lines {
+		if line != "" {
+			count++
+		}
+	}
+	return count
+}
+
+// GetPackageFiles retrieves the list of files installed by the package
+func GetPackageFiles(pkg string) ([]string, error) {
+	if err := IsPackageInstalled(pkg); err != nil {
+		return nil, err
+	}
+
+	cmd := exec.Command("apk", "info", "-qL", pkg)
+	output, err := cmd.Output()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get files for package %q: %w", pkg, err)
+	}
+	// Look Again
+	return strings.Split(string(output), "\n"), nil
+}
+
+// IsEmptyPackage checks if the package is empty and only contains SBOM Files
+func IsEmptyPackage(pkg string) (bool, error) {
+	files, err := GetPackageFiles(pkg)
+	if err != nil {
+		return false, err
+	}
+
+	nonSBOMFileCount := 0
+	for _, file := range files {
+		if !strings.Contains(file, "var/lib/db/sbom") && !strings.HasSuffix(file, ".spdx.json") {
+			nonSBOMFileCount++
+		}
+	}
+
+	return nonSBOMFileCount == 0, nil
+}
+
+// GetPackageDescription retrieves the package description
+func GetPackageDescription(pkg string) (string, error) {
+	if err := IsPackageInstalled(pkg); err != nil {
+		return "", err
+	}
+
+	cmd := exec.Command("apk", "info", "--installed", "--description", pkg)
+	output, err := cmd.Output()
+	if err != nil {
+		return "", fmt.Errorf("failed to get description for package %q: %w", pkg, err)
+	}
+
+	lines := strings.Split(strings.TrimSpace(string(output)), "\n")
+	if len(lines) < 2 {
+		return "", fmt.Errorf("unexpected description format for package %s", pkg)
+	}
+	return strings.TrimSpace(lines[1]), nil
+}
+
+func GetPackageDependency(pkg string) ([]string, error) {
+	if err := IsPackageInstalled(pkg); err != nil {
+		return nil, err
+	}
+
+	cmd := exec.Command("apk", "info", "--installed", "--depends", pkg)
```

Comment:
I would prefer not forking to apk, but using apko as a go library to read the apk database.
That path means that at some point in the future, we don't actually have to have apk installed in our test environment.

https://github.com/chainguard-dev/tw/pull/125 was an attempt to provide an 'apk free' test environment for shell scripts.  I abandoned it as apk is force-included in the test environments.